go/vt/tabletserver/tabletserver.go

```diff
@@ -233,7 +240,20 @@ func (tsv *TabletServer) StartService(target querypb.Target, dbconfigs dbconfigs
 	if err != nil {
 		return err
 	}
-	return tsv.SetServingType(tabletType, true)
+	return tsv.SetServingType(tabletType, true, nil)
+}
+
+// EnterLameduck causes tabletserver to enter the lameduck state. This
+// state causes health checks to fail, but the behavior of tabletserver
+// otherwise remains the same. Any subsequent calls to SetServingType will
+// cause the tabletserver to exit this mode.
+func (tsv *TabletServer) EnterLameduck() {
+	tsv.lameduck.Set(1)
+}
```

Comment:
If you look at my last change, you'll see that I initially had it as LAMEDUCK. I had also renamed one of the states to NOT_CONNECTED. But some tests started failing. So, I thought it was better not to be too granular.
Is there a benefit to someone knowing that a server is in lameduck mode?
Basically, I'm going by the precedent that other google servers don't reveal this state.